File: ios_monytix/ios_monytix/Components/WelcomeBanner.swift
//
//  WelcomeBanner.swift
//  ios_monytix
//
//  Created by santosh on 05/01/26.
//

import SwiftUI

struct WelcomeBanner: View {
    let username: String?
    
    private var displayName: String {
        if let username = username {
            // Extract name from email (part before @) or use full email
            let name = username.components(separatedBy: "@").first ?? username
            return name.capitalized
        }
        return "User"
    }
    
    var body: some View {
        HStack(spacing: 16) {
            // Bar chart icon
            ZStack {
                Circle()
                    .fill(Color.white.opacity(0.2))
                    .frame(width: 48, height: 48)
                
                Image(systemName: "chart.bar.fill")
                    .font(.system(size: 24, weight: .semibold))
                    .foregroundColor(.white)
            }
            
            // Text content
            VStack(alignment: .leading, spacing: 4) {
                Text("See where your money really goes.")
                    .font(.system(size: 16, weight: .bold))
                    .foregroundColor(.white)
                
                Text("Automatically categorized insights. Welcome back, \(displayName)!")
                    .font(.system(size: 13, weight: .regular))
                    .foregroundColor(.white.opacity(0.9))
                    .lineLimit(2)
            }
            
            Spacer()
        }
        .padding(.horizontal, 20)
        .padding(.vertical, 16)
        .background(
            Theme.goldGradient
        )
        .cornerRadius(16)
        .shadow(color: Color.black.opacity(0.2), radius: 8, x: 0, y: 4)
        .padding(.horizontal, 16)
        .padding(.vertical, 12)
    }
}

#Preview {
    VStack {
        WelcomeBanner(username: "santosh@example.com")
        WelcomeBanner(username: nil)
    }
    .background(Color(red: 0.18, green: 0.18, blue: 0.18))
}

File: apk_monytix/app/src/main/java/com/example/apk/ui/components/WelcomeBanner.kt
package com.example.apk.ui.components

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.BarChart
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Brush
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.example.apk.ui.theme.TextPrimary

@Composable
fun WelcomeBanner(
    username: String? = null,
    modifier: Modifier = Modifier
) {
    val displayName = if (username != null) {
        username.split("@").firstOrNull()?.replaceFirstChar { it.uppercaseChar() } ?: username
    } else {
        "User"
    }
    
    Row(
        modifier = modifier
            .fillMaxWidth()
            .padding(horizontal = 16.dp, vertical = 12.dp)
            .clip(RoundedCornerShape(16.dp))
            .background(
                Brush.horizontalGradient(
                    colors = listOf(
                        Color(0xFFFFD700), // Gold/Yellow
                        Color(0xFF8B5CF6)  // Purple
                    )
                )
            )
            .padding(horizontal = 20.dp, vertical = 16.dp),
        verticalAlignment = Alignment.CenterVertically,
        horizontalArrangement = Arrangement.spacedBy(16.dp)
    ) {
        // Bar chart icon
        Box(
            modifier = Modifier
                .size(48.dp)
                .clip(CircleShape)
                .background(Color.White.copy(alpha = 0.2f)),
            contentAlignment = Alignment.Center
        ) {
            androidx.compose.material3.Icon(
                imageVector = Icons.Default.BarChart,
                contentDescription = null,
                tint = Color.White,
                modifier = Modifier.size(24.dp)
            )
        }
        
        // Text content
        Column(
            modifier = Modifier.weight(1f),
            verticalArrangement = Arrangement.spacedBy(4.dp)
        ) {
            Text(
                text = "See where your money really goes.",
                fontSize = 16.sp,
                fontWeight = FontWeight.Bold,
                color = TextPrimary
            )
            Text(
                text = "Automatically categorized insights. Welcome back, $displayName!",
                fontSize = 13.sp,
                fontWeight = FontWeight.Normal,
                color = TextPrimary.copy(alpha = 0.9f),
                maxLines = 2
            )
        }
    }
}

